pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -659,29 +661,74 @@ type InterfaceTarget struct {
 }
 
 type Alias struct {
+	name        string
+	userDefined bool
+}
+
+// Package private, responsible to interact with xml.Encoder and xml.Decoder
+type userAliasXML struct {
 	Name string `xml:"name,attr"`
 }
 
-type UserAlias Alias
+// Package private, responsible to interact with json.Marshal and json.Unmarshal
+type userAliasJSON struct {
+	Name        string
+	UserDefined bool
+}
 
 type Rom struct {
 	Enabled string `xml:"enabled,attr"`
 }
 
+func NewLibvirtManagedAlias(aliasName string) *Alias {
+	return &Alias{name: aliasName}
+}
+
+func NewUserDefinedAlias(aliasName string) *Alias {
+	return &Alias{name: aliasName, userDefined: true}
+}
+
+func (alias Alias) GetName() string {
+	return alias.name
+}
+
+func (alias Alias) GetUserDefined() bool {
```

Comment:
This should probably be `IsUserDefined`.

Reply:
Func-name changed at the def and use sites.